### PR TITLE
front: display simulation results only if a train is selected or get data is ready

### LINKED ISSUE #10031: My scenario tries to load some output table though i have no trains in it
### What happened?

I just created a scenario, so i have no trains in it, my the front end displays this:

Image

### What did you expect to happen?

The scenrio should be completely empty.

### How can we reproduce it (as minimally and precisely as possible)?

I don't really know how to reproduce this unfortunately, it just happened. I'm guessing it's some kind of cache problem but i'm really not sure.

Edit : apperently opening any new scenario does that.

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

9e75ce2

### PR DESCRIPTION
**Pull request by clarani**
closes #10031

### CODE REVIEW

File: front/src/applications/operationalStudies/views/SimulationResults.tsx [hunk 1]
```diff
@@ -116,6 +116,10 @@ const SimulationResults = ({
     }
   }, [extViewport]);
 
+  if (!selectedTrainSchedule && !projectionData) {
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/views/SimulationResults.tsx]: Should we be more restrictive ? Sometimes there is a bug where we see the simulation results but not the space time chart (and we need to change the projection in order to display it)
```suggestion
  if (!selectedTrainSchedule || !projectionData) {
```

**Review by SharglutDev — COMMENTED** [review]
Works great, just one question :)

**Review by theocrsb — APPROVED** [review]
LGTM and tested, but I agree with @SharglutDev :)

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 1 above)
Comment (clarani) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: Ah I had in mind that as the SpaceTimeChart does not depend on the selected train, it should not disappear if the simulation of the selectedTrain is loading.

But I can hide all the results below the SpaceTimeChart if we have no selected train ?

File: front/src/applications/operationalStudies/views/SimulationResults.tsx [hunk 2]
```diff
@@ -137,16 +141,16 @@ const SimulationResults = ({
...
       {/* SIMULATION : SPACE TIME CHART */}
-      <ResizableSection
-        height={manchetteWithSpaceTimeChartHeight}
-        setHeight={setManchetteWithSpaceTimeChartHeight}
-        minHeight={MANCHETTE_WITH_SPACE_TIME_CHART_DEFAULT_HEIGHT}
-      >
-        <div
-          className="simulation-warped-map d-flex flex-row align-items-stretch mb-2"
-          style={{ height: manchetteWithSpaceTimeChartHeight }}
+      {projectionData && projectionData.projectedTrains.length > 0 && (
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/views/SimulationResults.tsx]: Hmm with this condition, when we switch projection, the space taken by the space time disappears resulting with an unpleasant animation where the rest of the simulation goes up

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: Why not, but we shouldn't do that for "no projection" (see other comment)

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 2 above)
Comment (clarani) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: You're right, I revert my changes :+1:

**Review by SharglutDev — COMMENTED** [review]
If I delete the selected train (which is not the projected one), it seems there is no new one selected (or probably it goes on an invalid one) and we keep seeing the map/output table loader.


https://github.com/user-attachments/assets/2acba9f3-f186-46cd-8fb1-364e1f9ec057

**Comment by clarani** [comment]
You're right, I think it's fixed like this, what do you think ?

**Review by SharglutDev — APPROVED** [review]
Thank you for the last change, everything looks great !

**Pull request merged**